cpp/src/arrow/array/util.cc

```diff
@@ -347,215 +349,234 @@ static Result<std::shared_ptr<Scalar>> MakeScalarForRunEndValue(
   return std::make_shared<Int64Scalar>(run_end);
 }
 
-// get the maximum buffer length required, then allocate a single zeroed buffer
-// to use anywhere a buffer is required
 class NullArrayFactory {
  public:
-  struct GetBufferLength {
```

Comment:
Why delete this?

Reply:
I didn't really delete it, just refactored its logic into NullArrayFactory. This way the code which requests preallocated zero buffer is in the same function as the code which uses the zero buffer, which seemed more clear to me.